### PR TITLE
Bug fix: Completing Offline Contact does not re-render screen

### PR DESCRIPTION
**Pull request by GPaoloni**
Jira: https://bugs.benetech.org/browse/CHI-582

This PR:
- Fixes a misspelling.
- Avoid selecting a task if is one created in behalf (offline contact process).
- Re renders agent desktop each time Task View unmounts an offline contact form.

### CODE REVIEW

**Review by nickhurlburt — APPROVED** [review]
Looks good.  When ending an offline contact with no other tasks, it re-renders the NoTasksView correctly.

When ending an offline contact when there are other tasks, it is not auto-selecting another task, but rather keeps the page clear.  This is the normal functionality when closing an online contact as well, though, so I don't think that's a problem.
image

File: plugin-hrm-form/src/components/TaskView.tsx [hunk 1]
```diff
@@ -20,18 +21,28 @@ type Props = OwnProps & ConnectedProps<typeof connector>;
...
     if (shouldRecreateState) {
       recreateContactState(currentDefinitionVersion.tabbedForms)(task.taskSid);
     }
   }, [currentDefinitionVersion.tabbedForms, recreateContactState, shouldRecreateState, task.taskSid]);
 
+  // Force a re-render on unmount (temporary fix NoTaskView issue with Offline Contacts)
+  React.useEffect(() => {
+    return () => {
+      if (isOfflineContactTask(task)) reRenderAgentDesktop();
+    };
```
Comment (andvirga) [inline plugin-hrm-form/src/components/TaskView.tsx:35]: Is this equivalent to do? (I'm not getting why are you returning a function instead, but it's just curiosity):  
```javascript 
if (isOfflineContactTask(task)) { 
  reRenderAgentDesktop();
}
```

**Review by andvirga — APPROVED** [review]
Looks good, I've only one question ;)

File: plugin-hrm-form/src/components/TaskView.tsx (hunk 1 above)
Comment (GPaoloni) [reply plugin-hrm-form/src/components/TaskView.tsx:35]: The returned function is the [cleanup of the effect](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup). In this case serves the purpose of `componentWillUnmount` (or when the `task` dependency changes, but that's also triggering an unmount of `TaskView`).

**Pull request merged**